plugins/outputs/kairosdb/kairosdb_test.go

```diff
@@ -0,0 +1,204 @@
+package kairosdb
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+func TestCleanTags(t *testing.T) {
+	var tagtests = []struct {
+		ptIn    map[string]string
+		outTags map[string]string
+	}{
+		{
+			map[string]string{"one": "two", "three": "four"},
+			map[string]string{"one": "two", "three": "four"},
+		},
+		{
+			map[string]string{"aaa": "bbb"},
+			map[string]string{"aaa": "bbb"},
+		},
+		{
+			map[string]string{"Sp%ci@l Chars[": "g$t repl#ce)d"},
+			map[string]string{"Sp-ci-l_Chars_": "g-t_repl-ce_d"},
+		},
+		{
+			map[string]string{"μnicodε_letters": "okαy"},
+			map[string]string{"μnicodε_letters": "okαy"},
+		},
+		{
+			map[string]string{"n☺": "emojies☠"},
+			map[string]string{"n_": "emojies_"},
+		},
+		{
+			map[string]string{},
+			map[string]string{},
+		},
+	}
+	for _, tt := range tagtests {
+		tags := cleanTags(tt.ptIn)
+		if !reflect.DeepEqual(tags, tt.outTags) {
+			t.Errorf("\nexpected %+v\ngot %+v\n", tt.outTags, tags)
+		}
+	}
+}
+
+func TestBuildTagsTelnet(t *testing.T) {
+	var tagtests = []struct {
+		ptIn    map[string]string
+		outTags string
+	}{
+		{
+			map[string]string{"one": "two", "three": "four"},
+			"one=two three=four",
+		},
+		{
+			map[string]string{"aaa": "bbb"},
+			"aaa=bbb",
+		},
+		{
+			map[string]string{"one": "two", "aaa": "bbb"},
+			"aaa=bbb one=two",
+		},
+		{
+			map[string]string{},
+			"",
+		},
+	}
+	for _, tt := range tagtests {
+		tags := ToLineFormat(tt.ptIn)
+		if !reflect.DeepEqual(tags, tt.outTags) {
+			t.Errorf("\nexpected %+v\ngot %+v\n", tt.outTags, tags)
+		}
+	}
+}
+
+func TestSanitize(t *testing.T) {
+	tests := []struct {
+		name     string
+		value    string
+		expected string
+	}{
+		{
+			name:     "Ascii letters and numbers allowed",
+			value:    "ascii 123",
+			expected: "ascii_123",
+		},
+		{
+			name:     "Allowed punct",
+			value:    "-_./",
+			expected: "-_./",
+		},
+		{
+			name:     "Special conversions to hyphen",
+			value:    "@*%#$!",
+			expected: "-----_",
+		},
+		{
+			name:     "Unicode Letters allowed",
+			value:    "μnicodε_letters",
+			expected: "μnicodε_letters",
+		},
+		{
+			name:     "Other Unicode not allowed",
+			value:    "“☢”",
+			expected: "___",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			actual := sanitize(tt.value)
+			require.Equal(t, tt.expected, actual)
+		})
+	}
+}
+
```

Comment:
Please add a test against a http-test server and check if the received metric adhere's to what you expect!